packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/AmplifyDataStorePlugin.kt

```diff
@@ -286,6 +288,9 @@ class AmplifyDataStorePlugin : FlutterPlugin, MethodCallHandler {
             schema = modelProvider.modelSchemas()[modelName]!!
             serializedModelData =
                 deserializeNestedModel(request["serializedModel"].safeCastToMap()!!, schema)
+
+            queryPredicate = QueryPredicateBuilder.fromSerializedMap(
+            request["queryPredicate"].safeCastToMap()) ?: QueryPredicates.all()
```

Comment:
*Question:* This could also default to null, correct?

I think it is fine how it is. Just making sure I understand the behavior?

Reply:
I'm curious if it *could* default to null and result in the same behavior, not if it *has to* be null. As long as it *can* default to null, you could make it the same on both platforms. That would be my preference, but this works as well.